### ISSUE TITLE
[Flutter Web] Tab key doesn't switch fields correctly

### ISSUE THREAD

**Issue by pieropetralla90** [flutter/flutter#113391]
Thanks to anyone who will take the time for me.

I have an application with various forms with textfields. In each form i noted that when i use tab button to switch fields, focus disappear from the fields and goes i dont know where.

I tried to use Focus widget to see if on tab the focus change, and it doesn't (only when the focus return to the first field, Focus return me that it is changed)

Below, a gift showing the unwanted behavior
Someone could help me? Thanks you

This is a snippet of Login form.

```
Focus(
     onFocusChange: ((value) => print("focus is changed")),
                      child: Column(
                        children: [
                          Semantics(
                            value: "Email",
                            child: TextFormField(
                              key: Key("Email"),
                              validator: (value) => emailValidator(value),
                              controller: _emailController,
                              
                            ),
                          ),
                          const SizedBox(height: 20),
                          Semantics(
                            value: "Password",
                            child: TextFormField(
                              key: Key("Password"),
                              validator: (value) =>
                                  formRequiredValidation("password", value),
                              textInputAction: TextInputAction.done,
                              obscureText: _isTypePassword,
                              controller: _passwordController,
                              onChanged: (value) {
                                setState(() {});
                              },
                              onFieldSubmitted: (String value) => _onSubmit(),
                              
                            ),
                          ),
                          const SizedBox(height: 20),
                          SizedBox(
                            width: double.infinity,
                            height: 45,
                            child: ElevatedButton(
                              key: Key("Accedi"),
                              onPressed: _onSubmit,
                              child: const Text(
                                "ACCEDI",
                                
                              ),
                            ),
                          ),
                          const SizedBox(height: 20),
                          GestureDetector(
                            onTap: () {},
                            child: Text(
                              "Hai dimenticato la password?",
                              style: TextStyle(
                                  color: Theme.of(context).primaryColor),
                            ),
                          ),
                          const SizedBox(height: 20),
                          GestureDetector(
                            onTap: () {},
                            child: Text(
                              "Non sei ancora registrato?",
                              style: TextStyle(
                                  color: Theme.of(context).primaryColor),
                            ),
                          ),
                        ],
                      ),
                    ),
```

![bug text tab](https://user-images.githubusercontent.com/51114737/195635627-f8f22e4d-b649-4a05-90e1-128aa751c6f7.gif)

**Comment by exaby73** [comment]
Hello @pieropetralla90 . Thank you for filing this issue. Can you please provide the following information?

- A [complete and minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Ideally, a single `main.dart` file that can be run directly.
- The output of `flutter doctor -v`

Please provide the information in the form of text. See how to make collapsible sections with Markdown [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections).

**Comment by pieropetralla90** [comment]
- Unfortunately I cannot provide a "A complete and minimal, reproducible example." because the problem occurs only in the forms inserted in the context of the APP.
- The output of flutter doctor -v
![flutterdoctor-v](https://user-images.githubusercontent.com/51114737/195662743-aa4a2433-3376-4ccd-b036-e0967ca1a125.png)

**Comment by pieropetralla90** [comment]
Added 3 important information:

- it only occurs in production and not in debug mode
- it only occurs on the web and not in iOS and Andorid
- we tried to compile for windows and it works fine, the problem does not occur

**Comment by exaby73** [comment]
Unfortunately, without a reproducible example, the issue is unactionable. If you can produce a sample that can be used to verify the issue, that would be great. If it cannot be reproduced, unfortunately, we cannot take the issue further.

**Comment by pieropetralla90** [comment]
ok, tomorrow I will prepare you a minimal repository with the main and the login form.
A thousand thanks

**Comment by CesareIsHere** [comment]
Hi, i'm working with @pieropetralla90 
Here the example repository:
[digiworks_debug_test.zip](https://github.com/flutter/flutter/files/9783922/digiworks_debug_test.zip)

I remembre you that the in debug mode through flutter server (flutter run web) the bug is not visible, so you need to use a web server like node or firebase.
Steps to reproduce the bug:
1) Open the application
2) Click tab to focus email textfield
3) Try to write something

Thank you very much!

**Comment by pieropetralla90** [comment]
Good morning , my colleague @CesareIsHere posted the minimal digiworks_debug_test.zip repository.
A thousand thanks 🙏

**Comment by exaby73** [comment]
Hello @pieropetralla90. Could you please tell me which browser you are testing on? Also, it would be great if you could include some code to set up a web server via node or firebase as you mentioned. Thank you.

**Comment by CesareIsHere** [comment]
Hi @exaby73.
Thank you for your attention but we've solved the issue!
Thank you again, bye!

**Comment by exaby73** [comment]
No issue. If you can, please do provide information about how you solved your issue so that if someone stumbles upon this in the future, your input may help them. Either way, I'll close this issue now. Thank you :)

**Comment by Digidevs** [comment]
Good morning,
we solved the problem by deleting the web folder and having it recreate in flutter.
We noticed that the new generated index.html file contains a third of the code compared to the old one.
It should be noted that the project started with Flutter 2 and was subsequently upgraded to Flutter 3.
Also note that with Flutter 3 important improvements have been released for the web part, including the ability to set up a loading page before starting the app or before loading it, along with other things.
Since it wasn't there in Flutter 2, we created a loading page ourselves.
it is our opinion that these changes resulted in the focus on objects not working.
Thanks to all for the precious collaboration.

**Comment by pieropetralla90** [comment]
Good morning,
we solved the problem by deleting the web folder and having it recreate in flutter.
We noticed that the new generated index.html file contains a third of the code compared to the old one.
It should be noted that the project started with Flutter 2 and was subsequently upgraded to Flutter 3.
Also note that with Flutter 3 important improvements have been released for the web part, including the ability to set up a loading page before starting the app or before loading it, along with other things.
Since it wasn't there in Flutter 2, we created a loading page ourselves.
it is our opinion that these changes resulted in the focus on objects not working.
Thanks to all for the precious collaboration.
